docker-java/src/test/java/com/github/dockerjava/cmd/EventsCmdIT.java

```diff
@@ -113,6 +114,31 @@ public void testEventStreamingWithFilter() throws Exception {
         }
     }
 
+    @Test
+    public void testEventStreamingWithEventTypeFilter() throws Exception {
+        assumeNotSwarm("", dockerRule);
+
+        String startTime = getEpochTime();
+        int expectedEvents = 1;
+
+        EventsTestCallback eventCallback = new EventsTestCallback(expectedEvents);
+
+        dockerRule.getClient().eventsCmd()
+            .withSince(startTime)
+            .withEventTypeFilter(EventType.CONTAINER)
+            .withEventFilter("start")
+            .exec(eventCallback);
+
+        generateEvents();
+
+        List<Event> events = eventCallback.awaitExpectedEvents(30, TimeUnit.SECONDS);
+
+        // we should only get "start" events here
+        for (Event event : events) {
+            assertThat("Received event: " + event, event.getAction(), is("start"));
```

Comment:
This test should check event's type, not the action. `generateEvents()` generates event of CONTAINER,IMAGE and NETWORK type. Consider filtering by "IMAGE", for example, and verifying that all events are of IMAGE type

Reply:
What happens when the event type is something that is not in EventType.   getType returns an enum class so this may cause issues.

Reply:
Guess it's fine since I am filtering it out on the test.  And users who use the getEventType would have to be cautious if there's a new event type added to the API.